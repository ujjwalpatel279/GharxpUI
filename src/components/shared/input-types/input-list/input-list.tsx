import React, { forwardRef, ChangeEvent } from 'react';
import { useForm } from 'react-hook-form';

import { InputSelect, InputRadio } from '..';

interface Options {
  id: number | string;
  value: string;
}

interface InputListProps {
  id: string;
  forceSelection?: boolean;
  legend: string;
  options: Options[];
  required?: boolean;
  disabled?: boolean;
  error?: boolean;
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string;
  inputListChangeHandler?: (e: ChangeEvent<HTMLInputElement | HTMLSelectElement>) => void;
}

type Refs = HTMLInputElement | HTMLSelectElement;

export const InputList = forwardRef<Refs, InputListProps>(
  (
    { id, forceSelection, legend, options, required, disabled, error, setValue, value, inputListChangeHandler },
    ref,
  ) => (
    <>
      {options?.length > 6 ? (
        <InputSelect
          id={id}
          options={options}
          forceSelection={forceSelection}
          required={required}
          error={error}
          ref={ref as React.Ref<HTMLSelectElement>}
          setValue={setValue}
          value={value}
          onChange={inputListChangeHandler}
          disabled={disabled}
        />
      ) : (
        <InputRadio
          name={id}
          legend={legend}
          options={options}
          ref={ref as React.Ref<HTMLInputElement>}
          setValue={setValue}
          value={value}
          onChange={inputListChangeHandler}
          disabled={disabled}
        />
      )}
    </>
  ),
);

InputList.displayName = 'InputList';
