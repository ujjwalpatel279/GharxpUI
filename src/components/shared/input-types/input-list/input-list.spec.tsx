import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputList } from './input-list';

describe('Input List Component', () => {
  afterEach(cleanup);

  it('renders the component', () => {
    const { container } = render(
      <InputList
        id="inputList"
        legend="Input list"
        required={false}
        options={[{ id: '1', value: 'test' }]}
        ref={() => null}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
