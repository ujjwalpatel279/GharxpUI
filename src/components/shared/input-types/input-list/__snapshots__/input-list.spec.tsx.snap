// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input List Component renders the component 1`] = `
<fieldset>
  <legend>
    Input list
  </legend>
  <div
    class="l-flex l-flex--wrap"
  >
    <div>
      <label
        class="c-input-list-item"
        for="1"
      >
        <input
          id="1"
          name="inputList"
          type="radio"
          value="test"
        />
        <span
          class="c-radio-input__inner"
        >
          <span>
            test
          </span>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;
